### PR TITLE
typeinferer code cleanup

### PR DESCRIPTION
**Pull request by fudali113**
Use const in type inferer's switch-case.
@shenli

### CODE REVIEW

**Comment by shenli** [comment]
@XuHuaiyu PTAL

**Comment by shenli** [comment]
@fudali113 Thanks for your PR!

File: plan/typeinferer.go [hunk 1]
```diff
@@ -335,99 +333,84 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
...
-		// The default return type of IF() (which may matter when it is stored into a temporary table) is calculated as follows.
-		// Expression	Return Value
-		// expr2 or expr3 returns a string	string
-		// expr2 or expr3 returns a floating-point value	floating-point
-		// expr2 or expr3 returns an integer	integer
-		tp = x.Args[1].GetType()
-	case "get_lock", "release_lock":
-		tp = types.NewFieldType(mysql.TypeLonglong)
-	case ast.AesEncrypt, ast.AesDecrypt, ast.SHA2, ast.InetNtoa:
+
```
Comment (XuHuaiyu) [inline plan/typeinferer.go]: Remove this line.

File: plan/typeinferer.go [hunk 2]
```diff
@@ -335,99 +333,84 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
...
+		tp = types.NewFieldType(mysql.TypeDuration)
+		tp.Decimal = v.getFsp(x)
+	case ast.Curdate, ast.CurrentDate, ast.Date, ast.FromDays:
+		tp = types.NewFieldType(mysql.TypeDate)
+	case ast.DateAdd, ast.DateSub, ast.AddDate, ast.SubDate, ast.Timestamp, ast.TimestampAdd, ast.StrToDate:
+		tp = types.NewFieldType(mysql.TypeDatetime)
+	case ast.Now, ast.Sysdate, ast.CurrentTimestamp, ast.UTCTimestamp:
+		tp = types.NewFieldType(mysql.TypeDatetime)
+		tp.Decimal = v.getFsp(x)
+
```
Comment (XuHuaiyu) [inline plan/typeinferer.go]: ditto.

File: plan/typeinferer.go [hunk 3]
```diff
@@ -335,99 +333,84 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
...
-		tp = aggFieldType(x.Args)
-		if tp.Tp == mysql.TypeVarchar {
-			tp.Tp = mysql.TypeVarString
-		}
-		classType := aggTypeClass(x.Args, &tp.Flag)
-		if classType == types.ClassString && !mysql.HasBinaryFlag(tp.Flag) {
-			tp.Charset, tp.Collate = types.DefaultCharsetForType(tp.Tp)
-		}
+		chs = v.defaultCharset
+
```
Comment (XuHuaiyu) [inline plan/typeinferer.go]: ditto.

File: plan/typeinferer.go (hunk 2 above)
Comment (XuHuaiyu) [inline plan/typeinferer.go]: ditto.

File: ast/functions.go [hunk 4]
```diff
@@ -164,52 +164,53 @@ const (
...
+	Trim            = "trim"
+	Upper           = "upper"
+	Ucase           = "ucase"
+	Hex             = "hex"
+	Unhex           = "unhex"
+	Rpad            = "rpad"
+	BitLength       = "bit_length"
+	CharFunc        = "char_func"
+	CharLength      = "char_length"
+	CharacterLength = "character_length"
```
Comment (coocood) [inline ast/functions.go]: We don't need to add it, in parser `character_length` is parsed to `char_length`.

File: plan/typeinferer.go [hunk 5]
```diff
@@ -335,99 +333,84 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
...
-		tp = types.NewFieldType(mysql.TypeDatetime)
-	case "dayname", "version", "database", "user", "current_user", "schema",
-		"concat", "concat_ws", "left", "right", "lcase", "lower", "repeat",
-		"replace", "ucase", "upper", "convert", "substring", "elt",
-		"substring_index", "trim", "ltrim", "rtrim", "reverse", "hex", "unhex",
-		"date_format", "rpad", "lpad", "char_func", "conv", "make_set", "oct", "uuid",
-		"insert_func", "bin", "quote", "format", "from_base64", "to_base64":
-		tp = types.NewFieldType(mysql.TypeVarString)
-		chs = v.defaultCharset
-	case "strcmp", "isnull", "bit_length", "char_length", "character_length", "crc32", "timestampdiff",
```
Comment (coocood) [inline plan/typeinferer.go]: "character_length" can be removed.

Comment (fudali113) [inline plan/typeinferer.go]: ok

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**